extensions/pkg/controller/healthcheck/general/managed_resource.go

```diff
@@ -82,11 +90,10 @@ func (healthChecker *ManagedResourceHealthChecker) Check(ctx context.Context, re
 	}, nil
 }
 
-func managedResourceIsHealthy(managedResource *resourcesv1alpha1.ManagedResource) (bool, *string, error) {
+func managedResourceIsHealthy(managedResource *resourcesv1alpha1.ManagedResource) (bool, error) {
 	if err := health.CheckManagedResource(managedResource); err != nil {
-		reason := "ManagedResourceUnhealthy"
-		err := fmt.Errorf("managed resource %s in namespace %s is unhealthy: %v", managedResource.Name, managedResource.Namespace, err)
-		return false, &reason, err
+		err := fmt.Errorf("managed resource %q in namespace %q is unhealthy: %v", managedResource.Name, managedResource.Namespace, err)
```

Comment:
nit: maybe I would make sense to use the new error syntax introduced with go1.13 (https://blog.golang.org/go1.13-errors). `%w` acts exactly like `%v` but the differnece that the initial error could be unwrapped if needed.
```suggestion
		err := fmt.Errorf("managed resource %q in namespace %q is unhealthy: %w", managedResource.Name, managedResource.Namespace, err)
```